backend/dataall/core/organizations/services/organization_service.py

```diff
@@ -0,0 +1,254 @@
+
+
+class OrganizationService:
+    """Service that serves request related to organization"""
+
+    @staticmethod
+    @has_tenant_permission(permissions.MANAGE_ORGANIZATIONS)
+    def create_organization(data):
+        context = get_context()
+        with context.db_engine.scoped_session() as session:
+            username = context.username
+            org = models.Organization(
+                label=data.get('label'),
+                owner=username,
+                tags=data.get('tags', []),
+                description=data.get('description', 'No description provided'),
+                SamlGroupName=data.get('SamlGroupName'),
+                userRoleInOrganization=OrganisationUserRole.Owner.value,
+            )
+            session.add(org)
+            session.commit()
+
+            member = models.OrganizationGroup(
+                organizationUri=org.organizationUri,
+                groupUri=data['SamlGroupName'],
+            )
+            session.add(member)
+
+            activity = Activity(
+                action='org:create',
+                label='org:create',
+                owner=username,
+                summary=f'{username} create organization {org.name} ',
+                targetUri=org.organizationUri,
+                targetType='org',
+            )
+            session.add(activity)
+
+            ResourcePolicy.attach_resource_policy(
+                session=session,
+                group=data['SamlGroupName'],
+                permissions=permissions.ORGANIZATION_ALL,
+                resource_uri=org.organizationUri,
+                resource_type=models.Organization.__name__,
+            )
+
+            return org
+
+    @staticmethod
+    @has_resource_permission(permissions.UPDATE_ORGANIZATION)
+    def update_organization(uri, data):
+        context = get_context()
+        with context.db_engine.scoped_session() as session:
+            organization = OrganizationRepository.get_organization_by_uri(session, uri)
+            for field in data.keys():
+                setattr(organization, field, data.get(field))
+            session.commit()
+
+            activity = Activity(
+                action='org:update',
+                label='org:create',
+                owner=context.username,
+                summary=f'{context.username} updated organization {organization.name} ',
+                targetUri=organization.organizationUri,
+                targetType='org',
+            )
+            session.add(activity)
+            ResourcePolicy.attach_resource_policy(
+                session=session,
+                group=organization.SamlGroupName,
+                permissions=permissions.ORGANIZATION_ALL,
+                resource_uri=organization.organizationUri,
+                resource_type=models.Organization.__name__,
+            )
+            return organization
+
+    @staticmethod
+    # TODO: resource permission check for permissions.GET_ORGANIZATION ?
```

Comment:
Let's add this resource permission check for `GET_ORGANIZATION`

Reply:
> Let's add this resource permission check for `GET_ORGANIZATION`


Tested in AWS Deploymnet (after addressing `TODO` comment and adding `GET_ORGANIZATION` resource permission check of `def get_organization(...)`
- [ ] Create organization
- [ ] Modify organization
- [ ] Get organization
- [ ] List organizations
- [ ] Invite group to organization
- [ ] List organization groups
- [ ] Remove group from organization
- [ ] Link environment to organization
- [ ] List organization environments
- [ ] Archive organization

Think we can add this line then good to approve on my end!

Reply:
Thanks a lot for deploying it! I added the permission check